src/main/scala/io/iohk/ethereum/db/dataSource/RocksDbDataSource.scala

```diff
@@ -24,8 +24,6 @@ class RocksDbDataSource(
     private var handles: Map[Namespace, ColumnFamilyHandle]
 ) extends DataSource {
 
-  private val logger = LoggerFactory.getLogger("rocks-db")
-
   @volatile
   private var isClosed = false
```

Comment:
WDYT about changing it to https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/atomic/AtomicReference.html
?

I have good experience with atomic refs. I remember using @volatile a very long time ago in Java and it was no reliable 🤔

Reply:
Or even https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/atomic/AtomicBoolean.html :)

TBH I think that volatile is enough here since updates of this var don't involve a read (and that's the basic problem classes in `java.util.concurrent.atomic` solve - making reads and writes atomically)

Reply:
as we are now writing to this variable only after write lock is in locked state,  this could probably be even a simple var without `@volatile` annotations

Reply:
I think that we need this variable as @volatile.
In case only one thread reads and writes the value of a volatile variable and other threads only read the variable, then the reading threads are guaranteed to see the latest value written to the volatile variable. 
I agree with @kapke, that volatile is enough here since updates of this var don't involve a read. Are we ok with this? Or do you think in another approach?

Reply:
but we only modify this variable from the inside of write lock and read only from inside read lock, which has strictly stronger guarantees than `volatile` variable. So the variable could be simple `var` and we would still have more synchronization than with `volatile` variable.

But to be honest we can leave it `volatile`.  (I am against atomics here as having cas operations inside locks, would really be an overkill)